plugins/contactpressuresensorsarray/include/yarp/dev/ContactPressureSensorsArrayDriver.h

```diff
@@ -0,0 +1,216 @@
+/*
+ * Copyright (C) 2018 Fondazione Istituto Italiano di Tecnologia RBCS & iCub Facility & ADVR
+ * Authors: see AUTHORS file.
+ * CopyPolicy: Released under the terms of the LGPLv2.1 or any later version, see LGPL.TXT or LGPL3.TXT
+ */
+
+#ifndef GAZEBOYARP_CONTACTPRESSURESENSORSARRAYDRIVER_H
+#define GAZEBOYARP_CONTACTPRESSURESENSORSARRAYDRIVER_H
+
+#include <yarp/dev/DeviceDriver.h>
+#include <yarp/dev/IAnalogSensor.h>
+#include <yarp/dev/PreciselyTimed.h>
+#include <yarp/os/Stamp.h>
+#include <yarp/os/Semaphore.h>
+
+#include <yarp/sig/Matrix.h>
+
+#include <gazebo/common/Plugin.hh>
+
+#include <gazebo/sensors/Sensor.hh>
+#include <gazebo/sensors/ContactSensor.hh>
+#include <gazebo/sensors/SensorManager.hh>
+
+#include <gazebo/physics/Contact.hh>
+#include <gazebo/physics/Link.hh>
+#include <gazebo/physics/physics.hh>
+
+#include <ignition/math/Pose3.hh>
+#include <ignition/math/Vector3.hh>
+#include <ignition/math/Matrix3.hh>
+#include <ignition/math.hh>
+
+namespace yarp
+{
+    namespace dev
+    {
+        class GazeboYarpContactPressureSensorsArrayDriver;
+    }
+}
+
+/**
+ * YARP Device Driver for an array of contact pressure sensors
+ * 
+ * This device is used to simulate the contact normal forces that might be 
+ * observed by an arbitrary number of load cells at specific locations with respect
+ * to a given link's origin frame. It uses the instance of a Gazebo Contact Sensor which
+ * provides contact information of the link (specifically providing wrenches acting at the CG). 
+ * This information is transformed and mapped to the desired contact normal forces.
+ * 
+ * In its current version, this device reads the location of the load cells (to be mentioned manually from the urdf or sdf)
+ * from a configuration file. Reads the contact forces and torques acting at the gazebo collision level through a 
+ * gazebo ContactSensor instance. Finds an equivalent wrench acting at the CG of the specified link. Transforms this
+ * equivalent wrench to the origin of the specified link. This transformation is necessary since the load cell locations are
+ * with respect to the link origin. Finally, the transformed equivalent wrench is mapped to an arrary of contact normal forces acting
+ * at the load cell locations. 
+ * 
+ * This mapping problem is an infinite solution problem and is solved by obtaining a pseudo inverse, 
+ * which gives the results with a minimized norm.
+ * 
+ * The configuation file must be specified in the sdf within the plugin element within the yarpConfigurationFile tags
+ * The configuration file must contain two groups [WRAPPER] and [DRIVER]
+ * Parameters accepted in the config argument of the open method for [WRAPPER]:
+ * | Parameter name| Type |Units|Default Value|Required |                 Description                |              Notes             |
```

Comment:
I am afraid we need a space between the line `Parameters accepted in the config argument of the open method for [WRAPPER]:` and `| Parameter name| Type |Units|Default Value|Required |                 Description                |              Notes             |`, otherwise the markdown table will not be correctly generated.